front_end/sdk/DartSupport.js

```diff
@@ -56,78 +58,72 @@ Dart.ToggleSourcemapsActionDelegate = class {
   }
 };
 
+/// Convert a Dart expression to JavaScript.
+///
+/// We wrap the expression in a function which provides the local scope
+/// variables. That lets us evaluate the expression as a standalone function and
+/// not try to evaluate in the right scope.
+///
+/// @param {SDK.ExecutionContext} executionContext
+/// @param {string} dartExpression
+/// @return {string} dartExpression compiled to JavaScript
+window.$dartExpressionFor = async function (executionContext, dartExpression) {
```

Comment:
why is this logic written in JavaScript? Seems better to write in Dart now that more logic is in dart.

Reply:
That would be nice, especially since it includes logic similar to what might be needed to implement VM protocol. But it's tricky, as this is called from devtools and runs inside it, doing a bunch of back and forth on the protocol. So it's hard to move it to the server,  and right now we don't have any Dart code included in DDT. I'm not sure where the ability to run Dart code living inside a JS main is, but it at least would be tricky. Added a TODO for it.

Reply:
sg